odoo/models.py

```diff
@@ -2438,8 +2456,112 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
                         # date/datetime field value to False.
                         row.setdefault('__range', {})[group] = False
 
+                elif field.type == "properties":
+                    additional_domain = self._read_group_format_properties_result(
+                        initial_rows,
+                        row,
+                        group,
+                        properties_definition,
+                    )
+
                 row['__domain'] = expression.AND([row['__domain'], additional_domain])
 
+    def _read_group_format_properties_result(self, data, row, group, definition):
+        """Modify the final read group properties result.
+
+        Replace the relational properties ids by a tuple with their display name,
+        replace the "raw" tags and selection values by a list containing their labels.
+        Adapt the domains for the Falsy group (we can't just keep (selection, =, False)
+        e.g. because some values in database might correspond to  option that have been
+        remove on the parent).
+
+        Return the domain to use to search the given group.
+        """
+        fullname = group.split(":")[0]
+        if '.' not in fullname:
+            raise ValueError('You must choose the property you want to group by.')
+
+        additional_domain = [(fullname, '=', row[group])]
+
+        property_type = definition.get('type')
+
+        if property_type == 'selection' and not row[fullname]:
+            # can not do ('selection', '=', False) because we might have
+            # option in database that does not exist anymore
+            options = definition.get('selection') or []
+            options = {option[0]: option[1] for option in options}
+            additional_domain = expression.OR([
+                [(fullname, '=', False)],
+                [(fullname, 'not in', tuple(options))],
+            ])
+
+        elif property_type == 'many2one':
+            all_groups = tuple(
+                line[fullname] if isinstance(line[fullname], int)
+                else line[fullname][0]
+                for line in data if line[fullname]
+            )
+            if not row[fullname]:
+                # can not do ('many2one', '=', False) because we might have
+                # record in database that does not exist anymore
+                additional_domain = expression.OR([
+                    [(fullname, '=', False)],
+                    [(fullname, 'not in', all_groups)],
+                ])
+
+            elif (comodel := definition.get('comodel')) in self.env:
+                records = self.env[comodel].browse(all_groups)
+                name_gets = {record.id: (record.id, record.display_name) for record in records}
+                row[fullname] = name_gets.get(row[fullname])
+
+        elif property_type == 'tags':
+            tags = definition.get('tags') or []
+            tags = {tag[0]: tag for tag in tags}
+            # use the "in" for the domain and not "="
```

Comment:
It come from the search PR, we need the `IN` SQL operator, but for that we need to know the property type, which required a SQL query

Because of that we just used the `in` domain operator (so we know we should use the `IN` SQL operator)